server/index.js

```diff
@@ -151,6 +150,33 @@ app.post('/api/cart', (req, res, next) => {
 
 });
 
+app.post('/api/orders', (req, res, next) => {
+  const { cartId } = req.session;
+  const { name, creditCard, shippingAddress } = req.body;
+
+  if (!name || !creditCard || !shippingAddress) {
+    res.status(400).json('You must supply a valid name, credit card, and shipping address');
+  }
+
+  if (!cartId) {
+    res.status(400).json('This is not your cart');
+  }
```

Comment:
Be sure to `return` if you don't want this middleware function to keep running. Otherwise your server may try to respond more than once and crash.